libpod/image/prune.go

```diff
@@ -87,6 +83,19 @@ func (ir *Runtime) GetPruneImages(all bool, filterFuncs []ImageFilter) ([]*Image
 				pruneImages = append(pruneImages, i)
 			}
 		}
+
+		//skip the cache or intermediate images
+		cache, err := i.Intermediate(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if cache {
+			continue
+		}
+
+		if i.Dangling() {
+			pruneImages = append(pruneImages, i)
```

Comment:
`pruneImages` may now contain the same image twice if `all == true`.

(Non-blocking: Overall I’d prefer the control flow to cleanly separate the decision (“should this be pruned?”) from the consequence (“add to `pruneImages`”), not intermix them on the various code paths.)

Reply:
This is also fixed now.